lib/puppet_metadata/metadata.rb

```diff
@@ -144,6 +144,44 @@ def add_supported_operatingsystems(at = nil, desired_os = nil)
       added
     end
 
+    # @param [Date] at The date to check the EOL time. Today is used when nil.
+    # @param [String] desired_os the name of the operating system from metadata.json we want to update
```

Comment:
You should document it also accepts `nil`. See https://rubydoc.info/gems/yard/file/docs/GettingStarted.md#declaring-types
```suggestion
    # @param [Date, nil] at The date to check the EOL time. Today is used when nil.
    # @param [String, nil] desired_os the name of the operating system from metadata.json we want to update
```